Lib/test/test_zipfile.py

```diff
@@ -2283,35 +2283,54 @@ def test_from_file(self):
         zi = zipfile.ZipInfo.from_file(__file__)
         self.assertEqual(posixpath.basename(zi.filename), 'test_zipfile.py')
         self.assertFalse(zi.is_dir())
+        self.assertFalse(zi.is_sym())
+        self.assertTrue(zi.is_reg())
         self.assertEqual(zi.file_size, os.path.getsize(__file__))
 
     def test_from_file_pathlike(self):
         zi = zipfile.ZipInfo.from_file(pathlib.Path(__file__))
         self.assertEqual(posixpath.basename(zi.filename), 'test_zipfile.py')
         self.assertFalse(zi.is_dir())
+        self.assertFalse(zi.is_sym())
+        self.assertTrue(zi.is_reg())
         self.assertEqual(zi.file_size, os.path.getsize(__file__))
 
     def test_from_file_bytes(self):
         zi = zipfile.ZipInfo.from_file(os.fsencode(__file__), 'test')
         self.assertEqual(posixpath.basename(zi.filename), 'test')
         self.assertFalse(zi.is_dir())
+        self.assertFalse(zi.is_sym())
+        self.assertTrue(zi.is_reg())
         self.assertEqual(zi.file_size, os.path.getsize(__file__))
 
     def test_from_file_fileno(self):
         with open(__file__, 'rb') as f:
             zi = zipfile.ZipInfo.from_file(f.fileno(), 'test')
             self.assertEqual(posixpath.basename(zi.filename), 'test')
             self.assertFalse(zi.is_dir())
+            self.assertFalse(zi.is_sym())
+            self.assertTrue(zi.is_reg())
             self.assertEqual(zi.file_size, os.path.getsize(__file__))
 
     def test_from_dir(self):
         dirpath = os.path.dirname(os.path.abspath(__file__))
         zi = zipfile.ZipInfo.from_file(dirpath, 'stdlib_tests')
         self.assertEqual(zi.filename, 'stdlib_tests/')
         self.assertTrue(zi.is_dir())
+        self.assertFalse(zi.is_sym())
+        self.assertFalse(zi.is_reg())
         self.assertEqual(zi.compress_type, zipfile.ZIP_STORED)
         self.assertEqual(zi.file_size, 0)
 
+    def test_from_symlink(self):
+        self.addCleanup(unlink, TESTFN)
+        os.symlink(os.path.abspath(__file__), TESTFN)
+        zi = zipfile.ZipInfo.from_file(TESTFN, 'symlink', follow_symlinks=False)
```

Comment:
Can you add a test in which follow_symlinks is True?

Reply:
done in ca5e302